### DISCUSSION TITLE
Usage within VueJS - Module "stream" has been externalized

### DISCUSSION THREAD

**Discussion by tito** [fief-dev/fief#238] [Ideas]
Hi !

I am trying to use Fief under VueJs/Quasar in SPA mode.
Even if there is no tutorial about it, i tried to do:

```typescript
import { Fief } from '@fief/fief';
import { FiefAuth } from '@fief/fief/browser';

const fiefClient = new fief.Fief({
  baseURL: 'http://example.localhost:7001',
  clientId: 'privateclientid',
})
const fiefAuth = new FiefAuth(fiefClient);
```

But as soon as i try to import and use symbols from fief, i got this error:

```
vue-router.mjs:3441 Error: Module "stream" has been externalized for browser compatibility and cannot be accessed in client code.
    at Object.get (browser-external:stream:3:11)
    at index.mjs:11:25
```

Any idea ?

**Comment by fief-bailiff** [comment]
Hail, @tito 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by tito** [comment]
It look like this https://github.com/fief-dev/fief-js/blob/main/src/fetch/index.ts#L1C1-L2C1 make it browser import fail.
I can replicate the exception:
```typescript
import nodeFetch from 'node-fetch';
console.log(nodeFetch)
```

**Comment by frankie567** [comment]
Hmm, I see. Managing compatibility with both Node.js and browser is quite difficult. I'll see how it can be fixed.

I suppose the underlying bundler is Vite?

**Reply by tito** [reply]
Yes it is

**Comment by tito** [comment]
I worked to get it working by removing `node-fetch` completely, `getFetch` and using `cross-fetch` directly.
PR soon, but dunno how to adapt tests yet.

**Reply by mk53202** [reply]
@tito Can you post the relevant parts you got working with vue. I'm receiving the streams error as well and not sure what I can do to get Fife working with vue.js.

**Comment by ocryle** [comment]
any update on this one? or some workaround, trying to make it work with vue as well
this is a big blocker for my project atm and it's a bit frustrating

**Comment by frankie567** [comment]
I'm currently working on the JS SDK and will remove `node-fetch` dependency, which should hopefully fix the build error with Vite.

**Reply by frankie567** [reply]
I've released a beta version of the JS SDK. It should work correctly with Vite:

```sh
npm i @fief/fief@beta
```

Let me know if it works for you!

**Reply by tito** [reply]
Hi @frankie567, sorry for the long delay. I've been able to update my app and no longer required a custom made fief-js, your beta module works perfectly for me ! (VueJS 3 + Quasar 2.6 + Vite)

**Comment by molivramento** [comment]
@frankie567 
I couldn't make it work with Quasar, it's sad to use keycloak, I really wanted to stay in fief, I feel more comfortable...

Would it be great to have a guide for Vue JS? Normally I get along well with backend and devops, but I have to admit that my frontend skills are just enough to call API and render in the browser.

**Reply by frankie567** [reply]
I'm not a Vue JS connoisseur, but yes, having a VueJS walkthrough is definitely in my plans :)
